packages/devtools_app/lib/src/screens/memory/panes/chart/controller/android_chart_controller.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_shared/devtools_shared.dart';
+import 'package:flutter/foundation.dart';
+
+import '../../../../../shared/charts/chart_controller.dart';
+import '../../../../../shared/charts/chart_trace.dart' as trace;
+import '../../../shared/primitives/memory_timeline.dart';
+import '../data/charts.dart';
+
+class AndroidChartController extends ChartController {
+  AndroidChartController(
+    this.memoryTimeline, {
+    required this.paused,
+    List<int> sharedLabels = const <int>[],
+  }) : super(
+          name: 'Android',
+          sharedLabelTimestamps: sharedLabels,
+        );
+
+  ValueListenable<bool> paused;
+  MemoryTimeline memoryTimeline;
+
+  // TODO(terry): Only load max visible data collected, when pruning of data
```

Comment:
Is this TODO still valid or do we know what it's referring to? If not, we should just remove it.

Reply:
Yes, if we ever work on performance of chart, this may be helpful.